include/zephyr/drivers/can/can_mcan.h

```diff
@@ -127,6 +127,17 @@
 #define CAN_MCAN_PSR_ACT  GENMASK(4, 3)
 #define CAN_MCAN_PSR_LEC  GENMASK(2, 0)
 
+enum can_mcan_psr_lec {
+	CAN_MCAN_PSR_LEC_NO_ERROR    = 0,
+	CAN_MCAN_PSR_LEC_STUFF_ERROR = 1,
+	CAN_MCAN_PSR_LEC_FORM_ERROR  = 2,
+	CAN_MCAN_PSR_LEC_ACK_ERROR   = 3,
+	CAN_MCAN_PSR_LEC_BIT1_ERROR  = 4,
+	CAN_MCAN_PSR_LEC_BIT0_ERROR  = 5,
+	CAN_MCAN_PSR_LEC_CRC_ERROR   = 6,
+	CAN_MCAN_PSR_LEC_NO_CHANGE   = 7
+};
```

Comment:
Please avoid converting register bit definitions into enumerations. Nothing prevents multiple bits from being set.

Reply:
Last error code (LEC) is a 3bit integer.
I think an enumeration makes sense here, these are all the possible values